framework/selftest.cpp

```diff
@@ -926,6 +926,25 @@ static const kvm_config_t *selftest_kvm_config_real_16bit_fail()
 }
 #endif // __linux__ && x86-64
 
+static int selftest_inject_idle(struct test *test, int cpu)
+{
+    auto loop_start = std::chrono::steady_clock::now();
+    TEST_LOOP(test, 1) {
+        for (int i = 0; i < 10000; ++i) {
+            int a_variable = cpu * 2;
+            a_variable += 1;
+        }
+    }
+    auto loop_end = std::chrono::steady_clock::now();
+    auto elapsed_time = duration_cast<microseconds>(loop_end - loop_start);
+    auto sleep_duration = sApp->shmem->current_test_sleep_duration;
+
+    if ((elapsed_time > 1.01 * sleep_duration) || (elapsed_time < 0.99 * sleep_duration)) {
```

Comment:
The less-than is fine.

The more-than is likely going to be a source of flaky failures due to VMs in GitHub being slow for any reason. I think that's why the Windows run failed, not due to Windowsnesses. Also, it's a debugoptimized build, so the loop above will have been eliminated.